File: barricade-annotations/src/main/java/com/mutualmobile/barricade/annotation/Response.java
package com.mutualmobile.barricade.annotation;

/**
 * Annotation to declare a response for an endpoint. Each endpoint will typically have many of
 * these. It has sensible defaults for most params.
 */
public @interface Response {

  String fileName();

  int statusCode() default 200;

  String type() default "application/json";

  boolean isDefault() default false;
}

File: barricade-annotations/src/main/java/com/mutualmobile/barricade/annotation/Response.kt
package com.mutualmobile.barricade.annotation

/**
 * Annotation to declare a response for an endpoint. Each endpoint will typically have many of
 * these. It has sensible defaults for most params.
 */
annotation class Response(
    val fileName: String,
    val statusCode: Int = 200,
    val type: String = "application/json",
    val isDefault: Boolean = false
)
